### ISSUE TITLE
Inferenced map is totally innacurate

### ISSUE THREAD

**Issue by PixelG16** [PixelG16/BTYSTE-2025#8]
After using my inferencing program to predict if there was rhododendron in an image the result resembled perlin noise and was totally innacurate.

**Comment by PixelG16** [comment]
I think the issue might be down to the loss function I'm using, currently its MSE but MIOU might be more suited to this task, I might also try changing the activation functions from relu to hard-tanh.

**Comment by PixelG16** [comment]
# Persisting issue
I've started using MIOU now and have lowered the learning rate while running for longer. This is from my 20 epoch training cycle with 400 images each epoch. I want to test longer and shorter learning rates. In testing this reached a MIOU score of 67, seems accurate. There also seems to be a glitch in loading the original channels, I believe this is down to my original file for splitting up hte maps.

![Image](https://github.com/user-attachments/assets/19d2a79e-a016-44d9-8efd-506cf285fc83)



![Image](https://github.com/user-attachments/assets/0ee52b5e-0905-493c-9565-00a1e751ff08)


![Image](https://github.com/user-attachments/assets/20f448ab-8fd8-49bd-a053-de202031913c)

**Comment by PixelG16** [comment]
I've fixed the issue with the channels as indicated in issue #9 and can see much mroe promising results after 100 epochs of training with a lr of 1e-4.



![Image](https://github.com/user-attachments/assets/79c55a86-0c5d-4262-8ef0-0c595e00ea6d)

The colours are inverted in the predicted mask as the values of hte black area are higher adn the white in the ground truth are higher but it is certainly predicting something a bit more accurate. Normalization might be needed and a different accuracy metric that values boundaries higher.

**Comment by iancullinane06** [comment]
I built a composite loss function to value accurate boundaries as well as recognising when there was no rhododendron present(false positives) but it didn't work very well, I now plan on attempting to build a DeepLabV3+ model instead.

**Comment by iancullinane06** [comment]
I think I've found the issue at long last, I was randomly flipping the images and masks to augment images except I was accidentally flipping each individually instead of flipping the two equally.

Current model seems to be performing well



![Image](https://github.com/user-attachments/assets/cd3d6dc4-001e-484a-ac32-9e4cc4305bc1)

red is the newest model
